server/handler.go

```diff
@@ -13,32 +13,40 @@ import (
 
 // See: https://github.com/sourcegraph/go-langserver/blob/master/langserver/handler.go#L206
 func (self *Server) newHandler() jsonrpc2.Handler {
-	return newLSPHandler(
-		jsonrpc2.HandlerWithError(self.handle),
-	)
+	handler := jsonrpc2.HandlerWithError(self.handle)
+
+	// Create a handler that uses the server's concurrent methods configuration
+	return newLSPHandler(handler, func(method string) bool {
+		if self.Options != nil {
```

Comment:
I'd suggest passing in the predicate function, to keep the decision of which methods are concurrent outside of this module.

Reply:
Sure, thanks